chore(transform_conformance): do not extend plugins if current options has plugins

Dunqing opened the pull request:
I saw many tests were ignored even if they don't contain any unsupported plugins

Dunqing commented:
* **#6627** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6627?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented on tasks/coverage/src/babel/mod.rs:
```diff
@@ -167,12 +167,13 @@ impl Case for BabelCase {
-            .iter()
-            .filter_map(Value::as_str)
-            .any(|p| not_supported_plugins.contains(&p));
+        let has_not_supported_plugins = self.options.plugins.iter().any(|p| match p {
+            Value::String(plugin_name) => not_supported_plugins.contains(&plugin_name.as_str()),
+            Value::Array(a) => a.iter().any(|plugin_name| {
+                plugin_name.as_str().is_some_and(|n| not_supported_plugins.contains(&n))
+            }),
+            _ => false,
+        });
```
Good spot!

I think it's only the first element of the nested array which is the plugin name. e.g. [this one](https://github.com/babel/babel/blob/3bcfee232506a4cebe410f02042fb0f0adeeb0b1/packages/babel-plugin-transform-arrow-functions/test/fixtures/arrow-functions/self-referential/options.json):

```json
{
  "plugins": [
    ["transform-arrow-functions", { "spec": false }],
    "transform-function-name"
  ]
}
```

So this would do the trick:

```rs
Array(a) => a.first().is_some_and(|plugin_name| {
    plugin_name.as_str().is_some_and(|n| not_supported_plugins.contains(&n))
}),
```

Or, if we want to find out if there are any more configs which we're not understanding correctly:

```rs
let has_not_supported_plugins = self.options.plugins.iter().any(|p| {
    let plugin_name = match p {
        Value::String(plugin_name) => Some(plugin_name.as_str()),
        Value::Array(a) => a.first().and_then(|plugin_name| plugin_name.as_str()),
        _ => None,
    };
    let plugin_name = plugin_name.expect("Failed to parse plugins config");
    not_supported_plugins.contains(&plugin_name)
});
```

Dunqing replied:
Great, I prefer to second way

Dunqing commented:
### Merge activity

* **Oct 16, 10:06 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
chore(transform_conformance): do not extend plugins if current options has plugins (#6627)

I saw many tests were ignored even if they don't contain any unsupported plugins
